lib/public-stats.js

```diff
@@ -274,3 +275,79 @@ const updateRetrievalTimings = async (pgClient, committees) => {
     JSON.stringify(rows)
   ])
 }
+
+/**
+ * @param {pg.Client} pgClient
+ * @param {Iterable<Committee>} committees
+ * @param {(minerId: string, cid: string) => (string[] | undefined)} findDealClients
+ */
+const updateDailyClientRetrievalStats = async (pgClient, committees, findDealClients) => {
+  const clientRetrievalStats = new Map()
+  /** @type {Map<string, Array<Measurement>>} */
+  const clientMeasurements = new Map()
+  for (const c of committees) {
+    const { minerId, payload_cid: payloadCid } = c.retrievalTask
+
+    const clients = findDealClients(minerId, payloadCid)
+    if (!clients || !clients.length) {
+      console.warn(`Invalid retrieval task (${minerId}, ${payloadCid}): no deal clients found. Excluding the task from daily per-client stats.`)
+      Sentry.captureException(new Error('Invalid retrieval task: no deal clients found.'), {
+        extra: {
+          minerId,
+          payload_cid: payloadCid
+        }
+      })
+      continue
+    }
```

Comment:
Small nitpick, but could you include one line after end of each `if` or `for` block. There are quite many of them given that we have to map data back to client and lack of whitespaces makes it a bit hard to read thought.